src/dataflow/src/source/postgres.rs

```diff
@@ -43,11 +43,59 @@ pub struct PostgresSourceReader {
     lsn: PgLsn,
 }
 
+trait ErrorExt {
+    fn is_recoverable(&self) -> bool;
+}
+
+impl ErrorExt for tokio_postgres::Error {
+    fn is_recoverable(&self) -> bool {
+        match self.source() {
+            Some(err) => {
+                match err.downcast_ref::<DbError>() {
+                    Some(db_err) => {
+                        use Severity::*;
+                        // Connection and non-fatal errors
+                        db_err.code() == &SqlState::CONNECTION_EXCEPTION
```

Comment:
I think ```SQLCLIENT_UNABLE_TO_ESTABLISH_SQLCONNECTION``` should also be retryable.

Reply:
Can this come up during our replication usage? Looking at the source code of postgres it looks like it's when using FDW to for pg to connect to an other pg or with dblink

I do need to add TOO_MANY_CONNECTIONS though, I'll push this now

Reply:
Oh sorry, not all FDW codes have the FDW prefix apparently.

Reply:
I just see that we discussed this already on Slack, many apologies, I spaced out grepping for stuff.